### PR TITLE
Replace equalizer to display flex on cards

### PR DESCRIPTION
**Pull request by decabeza**
## References

This PR close https://github.com/consul/consul/issues/3092.

## Objectives

Replace Foundation `data-equalizer` to CSS `display: flex` on cards for homepage and custom pages avoiding use JS and prevent the layout break when the user uses the back button in their browser.

Also, move cards views to shared partials to `shared/_cards.html.erb` and `shared/_card.html.erb`.

## Visual Changes

All the cards maintain the same height per row independently of the content.

homepage_cards

### CODE REVIEW

File: app/views/pages/_card.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div id="<%= dom_id(card) %>"
-     class="card small-12 medium-<%= card.columns %> column margin-bottom end" data-equalizer-watch>
+     class="card small-12 medium-<%= card.columns %> column margin-bottom end">
```
Comment (javierm) [inline app/views/pages/_card.html.erb:2]: I see this partial and the one in `welcome/_card` are identical :thinking:. Would it make sense to use one file instead, under either `cards/_card` or `shared/_card`? That way we wouldn't have to change the code in two places :smile:.

File: app/views/welcome/_cards.html.erb [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
-  <% @cards.each do |card| %>
-    <%= render "card", card: card %>
-  <% end %>
+<div class="row">
+  <div class="cards-container">
+    <% @cards.each do |card| %>
+      <%= render "card", card: card %>
+    <% end %>
+  </div>
 </div>
```
Comment (javierm) [inline app/views/welcome/_cards.html.erb:9]: Maybe we could also extract this block into a partial in either `cards/_cards` or `shared/_cards` :thinking:. What do you think?

File: app/assets/stylesheets/layout.scss [hunk 3]
```diff
@@ -2788,11 +2788,18 @@ table {
...
   }
 
+  .cards-container {
+    display: flex;
+    flex-wrap: wrap;
+  }
+
   .card {
     border: 0;
+    flex-grow: 0;
```
Comment (javierm) [inline app/assets/stylesheets/layout.scss:2798]: AFAIK `flex-grow` is `0` by default :thinking:. Is there a reason to explicitely add this line?

File: app/views/pages/_card.html.erb (hunk 1 above)
Comment (decabeza) [reply app/views/pages/_card.html.erb:2]: Yes, one file it's better! I've created two new partials `shared/_cards` and `shared/_card`. 😌

**Pull request merged**